editor/filesystem_dock.cpp

```diff
@@ -1852,6 +1907,50 @@ void FileSystemDock::_overwrite_dialog_action(bool p_overwrite) {
 	_move_operation_confirm(to_move_path, to_move_or_copy, p_overwrite ? OVERWRITE_REPLACE : OVERWRITE_RENAME);
 }
 
+void FileSystemDock::_convert_dialog_action() {
+	Vector<String> valid_conversion_targets = _get_valid_conversions_for_file_paths(to_convert);
+
+	Vector<Ref<Resource>> selected_resources;
+	for (const String &S : to_convert) {
+		Ref<Resource> res = ResourceLoader::load(S);
+		ERR_FAIL_COND(!res.is_valid());
+		selected_resources.push_back(res);
+	}
+
+	Vector<Ref<Resource>> converted_resources;
+	for (Ref<Resource> res : selected_resources) {
+		Vector<Ref<EditorResourceConversionPlugin>> conversions = EditorNode::get_singleton()->find_resource_conversion_plugin_for_resource(res);
+		for (Ref<EditorResourceConversionPlugin> E : conversions) {
+			if (E->converts_to() == valid_conversion_targets[selected_conversion_id]) {
+				Ref<Resource> converted_res = E->convert(res);
+				ERR_FAIL_COND(!res.is_valid());
+				converted_resources.push_back(converted_res);
+
+				// Clear history for the objects being replaced.
+				EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+				undo_redo->clear_history(true, undo_redo->get_history_id_for_object(res.ptr()));
```

Comment:
This potentially clears history for the same context multiple times. You could have a HashSet of history ids that need clearing, to avoid repeated clears (it's not a big problem though).